tools/explorer/pkg/workloads/types/reservation.go

```diff
@@ -45,26 +46,23 @@ const (
 	Deleted = generated.NextActionDeleted
 )
 
-// ReservationQuery type to parse reservation query string
-type ReservationQuery struct {
-	CustomerID int
-	NextAction int
-}
-
 // Parse parese the query string
-func (rq *ReservationQuery) Parse(r *http.Request) mw.Response {
+func Parse(r *http.Request, filter ReservationFilter) (ReservationFilter, mw.Response) {
```

Comment:
Why this return a mw.Response and not an error? I can see you only return bad request 
so It's better to return an error, and in the handler if u receive an error u wrap it in the bad request response.

Reply:
no reaon at all corrected